dashboard/app/controllers/api/v1/users_controller.rb

```diff
@@ -45,7 +45,7 @@ def current
         in_section: current_user.student? ? current_user.sections_as_student.present? : nil,
         created_at: current_user.created_at,
         has_seen_ai_assessments_announcement: current_user.has_seen_ai_assessments_announcement?,
-        ai_differentiation_enabled: current_user.ai_differentiation_enabled?,
+        ai_differentiation_toggled_off: current_user.ai_differentiation_toggled_off?,
```

Comment:
Adopt the suggestions in this file and that removes the need for most of the frontend changes.

```suggestion
        ai_differentiation_enabled: !current_user.ai_differentiation_toggled_off?,
```